src/stdlib/models/Petri.jl

```diff
@@ -0,0 +1,42 @@
+module Petri
+export lens_dynamics
+
+using AlgebraicPetri
+using Catlab.CategoricalAlgebra
+
+using ..SimpleLenses
+using ...StdTheories: ThRing
+using ....Syntax
+using ....Util
+using ....Dsl
+
+CtxTrm(i) = Trm(Lvl(i;context=true)) # upstream this?
+
+function lens_dynamics(p::LabelledPetriNet, T::Type=ThRing)
+  i_Ob = lookup(gettheory(T), Default())
+  pos = Context([(Name(s),Typ(i_Ob)) for s in p[:sname]])
```

Comment:
Make an overload for `Context` that takes in a list of names and gives them all the default type.